src/plugin/track/trackinteraction.js

```diff
@@ -99,6 +101,40 @@ class TrackInteraction extends MeasureInteraction {
       this.config_.callback(all);
     }
   }
+
+  /**
+   * Either,
+   * 1) Immediately enable and begin TrackInteraction from the original right-click event, OR
+   * 2) Disable and deactivate the TrackInteraction
+   *
+   * @param {boolean} toggle
+   * @param {MapBrowserEvent=} opt_mapBrowserEvent
+   * @param {pluginx.track.TrackOptions=} opt_trackOptions starting point, callback, etc
+   */
+  trigger(toggle, opt_mapBrowserEvent, opt_trackOptions) {
+    if (toggle && opt_trackOptions) {
+      this.config(opt_trackOptions);
+    }
+
+    this.setEnabled(toggle);
+    this.setActive(toggle);
+
+    if (toggle && opt_mapBrowserEvent) {
+      let event;
+
+      // prime the coordinates of the pointerdown event for the DrawInteraction part of TrackInteraction
+      event = opt_mapBrowserEvent;
+      event['type'] = 'pointerdown';
+      event['originalEvent'] = new PointerEvent('pointerdown');
```

Comment:
Modifying these properties with bracket notation seems dangerous. Do these work in compiled mode? 

Would it be better to create a new `MapBrowserEvent` instance to pass to `handleEvent` below? Or do the modifications need to be on the original event reference for whatever else it does as part of the event handler chain?

Reply:
Yep; you're spot on about why I did it this way -- OpenLayers didn't seem to trust the `new ol.MapBrowserEvent()` when I tried it that way. Though I did make some tweaks here that made it work better in this version that I didn't try again with the `new` event version. Namely, I stopped passing a clone of the `mapBrowserEvent.originalEvent` to the `new PointerEvent()`'s optional second parameter. That parameter gave the `new PointerEvent()` the same positions and map information as the original event, but for some reason that made the whole outer MapBrowserEvent stop working with the top-level ol.Interaction classes.